packages/analytics-js-integrations/src/integrations/Optimizely/browser.js

```diff
@@ -112,28 +112,37 @@ class Optimizely {
 
   initOptimizelyIntegration(referrerOverride, sendCampaignData) {
     const newActiveCampaign = (id, referrer) => {
-      const state = window?.optimizely?.get('state');
-      if (state) {
-        const activeCampaigns = state.getCampaignStates({
-          isActive: true,
-        });
-        const campaignState = activeCampaigns[id];
-        if (referrer) campaignState.experiment.referrer = referrer;
-        sendCampaignData(campaignState);
+      try {
```

Comment:
can we check if reference of optimizely object can be avoided in init flow or whether this code is reachable ? 
The purpose of `initOptimizelyIntegration` itself is to initialise the object & we are trying to access the object within the init flow - are we missing something ?

Reply:
can do a check `if window.optimizely` instead

Reply:
It will provide ambiguous result. Because [here](https://github.com/rudderlabs/rudder-sdk-js/pull/1441) we are already assigning it as empty array. And `if window.optimizely` it will always return true.

Reply:
ok, we need code refactoring for this destination - many flaws.
If we had `state` in windows.optimizely object, this means it's a json object. Now in this case, when `registerFutureActiveCampaigns` is called, it adding an object which will definitely fail. Check if we need to add try-catch here too